src/query/Query.ts

```diff
@@ -1076,8 +1135,33 @@ export default class Query<T extends Model = Model> {
   /**
    * Convert given record to the model instance.
    */
-  hydrate (record: Data.Record): Data.Instance {
-    const model = this.model
+  hydrate (record: Data.Record, forceModel?: typeof Model): Data.Instance {
+
+    if(forceModel !== undefined) {
+      return new forceModel(record)
+    }
+
+    let model = this.model 
+
+    // If the record has the right typeKey attribute set, and Model has type mapping 
+    // we hydrate it as the corresponding model
+    if (record && record[model.typeKey] && Object.keys(model.types()).length > 0) {
```

Comment:
Could this be replaced by the `hasTypes()` method that I noticed above, for reducing code duplication?
```suggestion
    if (record && record[model.typeKey] && model.hasTypes()) {
```

Reply:
Thanks, just applied the change!